build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module.exports = function(babel) {
+  const {types: t} = babel;
+  return {
+    name: 'transform-amp-extension-call',
+    visitor: {
+      CallExpression(path) {
+        const {node} = path;
+        const {callee} = node;
+        if (t.isIdentifier(callee.object, {name: 'AMP'}) &&
+            t.isIdentifier(callee.property, {name: 'extension'})) {
+          const func = node.arguments[node.arguments.length - 1];
+          if (!(t.isFunctionExpression(func) ||
+              t.isArrowFunctionExpression(func))) {
+            return;
+          }
+
+          const newParams = [t.identifier('AMP')];
+          const newFunc = t.isFunctionExpression(func) ?
+              t.functionExpression(null, newParams, func.body) :
+              t.arrowFunctionExpression(newParams, func.body);
```

Comment:
@jridgewell i forgot to handle arrow expressions. added here and tests

Reply:
Why not just take invoke the original function? There's no need to create a new function node.

Reply:
See https://astexplorer.net/#/gist/ed168eb5b5098c06a751208884ebff46/7a5af9903baa4817f26d079bf720ae82ad3a2899

Reply:
thanks, done.  I wasn't sure if it was safe to reparent nodes like that.